direct/io-kafka/src/main/java/cz/o2/proxima/direct/kafka/KafkaLogReader.java

```diff
@@ -224,18 +224,20 @@ ObserveHandle processConsumer(
 
     final OffsetCommitter<TopicPartition> offsetCommitter = createOffsetCommitter();
 
-    BiConsumer<TopicPartition, ConsumerRecord<Object, Object>> preWrite =
-        (tp, r) ->
-            offsetCommitter.register(
-                tp,
-                r.offset(),
-                1,
-                () -> {
-                  OffsetAndMetadata mtd = new OffsetAndMetadata(r.offset() + 1);
-                  if (commitToKafka) {
-                    kafkaCommitMap.put(tp, mtd);
-                  }
-                });
+    final BiConsumer<TopicPartition, ConsumerRecord<Object, Object>> preWrite =
+        (tp, r) -> {
+          final long offset = r.offset();
+          offsetCommitter.register(
+              tp,
+              r.offset(),
```

Comment:
just offset?

Reply:
ehm, next time I should refresh page before sending comment ....

Reply:
the purpose of this change is that the inner lambda does not hold reference to ConsumerRecord (which can be potentially large as it contains serialized value + other metadata). This way we only hold reference for a single unboxed long value.